### ISSUE TITLE
sendtoaddress subtractfeefromamount=true does not respect paytxfee values

### ISSUE THREAD

**Issue by cryptogenesis** [bitcoin/bitcoin#10034]
When using subtractfeeamount=true with the sendtoaddress RPC call, the wallet does not always respect paytxfee values and seems to occasionally employ an arbitrary (and very high! 0.004 BTC/KB at latest glance) fee. Strangely, after sending a standard call with subtractfeeamount=false, subsequent calls that include subtractfeeamount=true seem to respect the value set by paytxfee.

**Comment by jonasschnelli** [comment]
Can you write down some steps to reproduce... a little test told me if working correct. But maybe I'm wrong.

Started `./src/bitcoind  -printtoconsole --regtest --paytxfee=100`
Sent with SFA: `./src/bitcoin-cli --regtest sendtoaddress n3NkSZqoPMCQN5FENxUBw4qVATbytH6FDK 1 "" "" true`
List tx tells me:
```
  "details": [
    {
      "account": "",
      "address": "n3NkSZqoPMCQN5FENxUBw4qVATbytH6FDK",
      "category": "send",
      "amount": -0.90000000,
      "label": "",
      "vout": 1,
      "fee": -0.10000000,
      "abandoned": false
    }
```

**Comment by cryptogenesis** [comment]
Hi Jonas,

I have a bash script pay.sh that just calls:

#!/bin/bash
bitcoin-cli sendtoaddress $1 $2 "" "" true

I then call this like ./pay.sh {address} {amount}

My current info is:

~/.bitcoin$ bitcoin-cli getinfo
{
  "version": 140000,
  "protocolversion": 70015,
  "walletversion": 130000,
  "balance": X.XXXXXXXX,
  "blocks": 457937,
  "timeoffset": 0,
  "connections": XX,
  "proxy": "",
  "difficulty": 475705205061.6292,
  "testnet": false,
  "keypoololdest": 1489899731,
  "keypoolsize": 100,
  "paytxfee": 0.0016XXXX,
  "relayfee": 0.00005000,
  "errors": ""
}

I routinely empty this wallet. I also have a CRON running on the hour that adjusts fees to target 5 block confirmation:

#!/bin/bash
FEE="$(bitcoin-cli estimatefee 5)"
bitcoin-cli settxfee $FEE

However, when I call pay.sh for the first time after receiving a deposit, it doesn't seem to respect the value of paytxfee (and has been paying 0.004 BTC/KB). I noticed that if I call the sendtoaddress command without referencing subtractfeefromamount, and then call my pay.sh script, the subtractfeefromamount will then reference the correctly defined paytxfee. Very odd to say the least.

I am happy to provide txids privately if you'd like which will demonstrate the behavior above.

Thank you for taking a look.

**Comment by MarcoFalke** [comment]
If this only and always happens the fist time you send, there might be a bug in your code to set fees. Otherwise, this is might be an instance of the known issue: https://github.com/bitcoin/bitcoin/blob/baae3149d63909c527e7b28c52c9124957ce8206/src/wallet/wallet.cpp#L2567

**Comment by morcos** [comment]
@cryptogenesis sorry for the delay, but i'm going to take another look at this.
If you'd be willing to point me to a few txid's, i'm not sure if it will help or not

morcos at chaincode.com

Thanks

**Comment by morcos** [comment]
@cryptogenesis I opened #10942 to fix a (hopefully very rare) edge case that was not yet fixed.  But I'm not sure that's the cause of your issue.

You are sure that the estimatefee call wasn't sometimes causing you to settxfee to a higher feerate?

I looked through the code and didn't see anything obviously wrong, but am willing to look more if you have more information (I realize its quite old now).

**Comment by MarcoFalke** [comment]
Most likely fixed in #10942 by @morcos!
